File: PAMO-Kotlin/BMI_Calculator/app/src/main/java/com/example/bmi_calculator/activity_quiz.java
package com.example.bmi_calculator;

import androidx.appcompat.app.AppCompatActivity;

import android.app.AlertDialog;
import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Random;
import java.util.Set;

public class activity_quiz extends AppCompatActivity {

    Button answer1, answer2, answer3, answer4;

    TextView score, question;

    private Questions mQuestions = new Questions();

    private String mAnswer;
    private int mScore = 0;
    private int questionNumber = 0;
    private int mQuestionsLength = mQuestions.mQuestions.length;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_quiz);

        answer1 = (Button) findViewById(R.id.answer1);
        answer2 = (Button) findViewById(R.id.answer2);
        answer3 = (Button) findViewById(R.id.answer3);
        answer4 = (Button) findViewById(R.id.answer4);

        score = (TextView) findViewById(R.id.score);
        question = (TextView) findViewById(R.id.question);

        score.setText("Score: " + mScore);

        updateQuestion(questionNumber);

        answer1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view){
                if(answer1.getText() == mAnswer){
                    mScore++;
                    updateQuestion(questionNumber);
                }
                else{
                    gameOver();
                }
            }
        });

        answer2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view){
                if(answer2.getText() == mAnswer){
                    mScore++;
                    updateQuestion(questionNumber);
                }
                else{
                    gameOver();
                }
            }
        });

        answer3.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view){
                if(answer3.getText() == mAnswer){
                    mScore++;
                    updateQuestion(questionNumber);
                }
                else{
                    gameOver();
                }
            }
        });

        answer4.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view){
                if(answer4.getText() == mAnswer){
                    mScore++;
                    updateQuestion(questionNumber);
                }
                else{
                    gameOver();
                }
            }
        });
    }

    private void updateQuestion(int num){
        if (questionNumber < mQuestionsLength){
            score.setText("Score: " + mScore);
            question.setText(mQuestions.getQuestion(num));
            answer1.setText(mQuestions.getChoice1(num));
            answer2.setText(mQuestions.getChoice2(num));
            answer3.setText(mQuestions.getChoice3(num));
            answer4.setText(mQuestions.getChoice4(num));

            mAnswer = mQuestions.getCorrectAnswer(num);
            questionNumber++;
        }
        else {
            score.setText("Score: " + mScore);
            gameWon();
        }
    }

    private void gameOver(){
        AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(activity_quiz.this);
        alertDialogBuilder
                .setMessage("You lose! Your score is " + mScore + " points.")
                .setCancelable(false)
                .setPositiveButton("NEW GAME",
                        new DialogInterface.OnClickListener() {
                            @Override
                            public void onClick(DialogInterface dialogInterface, int i) {
                                startActivity(new Intent(getApplicationContext(), activity_quiz.class));
                            }
                        })
                .setNegativeButton("EXIT",
                        new DialogInterface.OnClickListener(){
                            @Override
                            public void onClick(DialogInterface dialogInterface, int i){
                                finish();
                            }
                        });
        AlertDialog alertDialog = alertDialogBuilder.create();
        alertDialog.show();
    }

    private void gameWon(){
        AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(activity_quiz.this);
        alertDialogBuilder
                .setMessage("You won! Your score is " + mScore + " points.")
                .setCancelable(false)
                .setPositiveButton("NEW GAME",
                        new DialogInterface.OnClickListener() {
                            @Override
                            public void onClick(DialogInterface dialogInterface, int i) {
                                startActivity(new Intent(getApplicationContext(), activity_quiz.class));
                            }
                        })
                .setNegativeButton("EXIT",
                        new DialogInterface.OnClickListener(){
                            @Override
                            public void onClick(DialogInterface dialogInterface, int i){
                                finish();
                            }
                        });
        AlertDialog alertDialog = alertDialogBuilder.create();
        alertDialog.show();
    }
}

File: PAMO-Kotlin/BMI_Kotlin/app/src/main/java/com/example/bmi_calculator/activity_quiz.kt
package com.example.bmi_calculator

import android.app.AlertDialog
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.example.bmi_calculator.activity_quiz

class activity_quiz : AppCompatActivity() {
    var answer1: Button? = null
    var answer2: Button? = null
    var answer3: Button? = null
    var answer4: Button? = null
    var score: TextView? = null
    var question: TextView? = null
    private val mQuestions = Questions()
    private var mAnswer: String? = null
    private var mScore = 0
    private var questionNumber = 0
    private val mQuestionsLength = mQuestions.mQuestions.size
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_quiz)
        answer1 = findViewById<View>(R.id.answer1) as Button
        answer2 = findViewById<View>(R.id.answer2) as Button
        answer3 = findViewById<View>(R.id.answer3) as Button
        answer4 = findViewById<View>(R.id.answer4) as Button
        score = findViewById<View>(R.id.score) as TextView
        question = findViewById<View>(R.id.question) as TextView
        score!!.text = "Score: $mScore"
        updateQuestion(questionNumber)
        answer1!!.setOnClickListener {
            if (answer1!!.text === mAnswer) {
                mScore++
                updateQuestion(questionNumber)
            } else {
                gameOver()
            }
        }
        answer2!!.setOnClickListener {
            if (answer2!!.text === mAnswer) {
                mScore++
                updateQuestion(questionNumber)
            } else {
                gameOver()
            }
        }
        answer3!!.setOnClickListener {
            if (answer3!!.text === mAnswer) {
                mScore++
                updateQuestion(questionNumber)
            } else {
                gameOver()
            }
        }
        answer4!!.setOnClickListener {
            if (answer4!!.text === mAnswer) {
                mScore++
                updateQuestion(questionNumber)
            } else {
                gameOver()
            }
        }
    }

    private fun updateQuestion(num: Int) {
        if (questionNumber < mQuestionsLength) {
            score!!.text = "Score: $mScore"
            question!!.text = mQuestions.getQuestion(num)
            answer1!!.text = mQuestions.getChoice1(num)
            answer2!!.text = mQuestions.getChoice2(num)
            answer3!!.text = mQuestions.getChoice3(num)
            answer4!!.text = mQuestions.getChoice4(num)
            mAnswer = mQuestions.getCorrectAnswer(num)
            questionNumber++
        } else {
            score!!.text = "Score: $mScore"
            gameWon()
        }
    }

    private fun gameOver() {
        val alertDialogBuilder = AlertDialog.Builder(this@activity_quiz)
        alertDialogBuilder
                .setMessage("You lose! Your score is $mScore points.")
                .setCancelable(false)
                .setPositiveButton("NEW GAME"
                ) { dialogInterface, i -> startActivity(Intent(applicationContext, activity_quiz::class.java)) }
                .setNegativeButton("EXIT"
                ) { dialogInterface, i -> finish() }
        val alertDialog = alertDialogBuilder.create()
        alertDialog.show()
    }

    private fun gameWon() {
        val alertDialogBuilder = AlertDialog.Builder(this@activity_quiz)
        alertDialogBuilder
                .setMessage("You won! Your score is $mScore points.")
                .setCancelable(false)
                .setPositiveButton("NEW GAME"
                ) { dialogInterface, i -> startActivity(Intent(applicationContext, activity_quiz::class.java)) }
                .setNegativeButton("EXIT"
                ) { dialogInterface, i -> finish() }
        val alertDialog = alertDialogBuilder.create()
        alertDialog.show()
    }
}